lib/ansible/modules/windows/win_partition.ps1

```diff
@@ -0,0 +1,317 @@
+                        Fail-Json -obj $result -message "Unable to change partition size: $($_.Exception.Message)"
+                    }                    
+                    $result.changed = $true
+                }
+            } elseif ($ansible_partition_size -gt $max_supported_size) {
+                Fail-Json -obj $result -message "Specified partition size exceeds size supported by partition"
+            }
+        }
+
+        if ($drive_letter -NotIn ("auto", $null, $ansible_partition.DriveLetter)) {
+            if (-not $check_mode) {
+                try {
+                    Set-Partition -DiskNumber $ansible_partition.DiskNumber -PartitionNumber $ansible_partition.PartitionNumber -NewDriveLetter $drive_letter
+                } catch {
+                    Fail-Json -obj $result -message "Unable to change drive letter: $($_.Exception.Message)"
+                }
+            }
+            $result.changed = $true
+        }
+    }
+}
+else {
+    if ($state -eq "present") {
+        if ($null -eq $disk_number) {
+            Fail-Json -obj $result -message "Missing required parameter: disk_number"
+        }
+        if ($null -eq $ansible_partition_size -and -not $size_is_maximum){
+            Fail-Json -obj $result -message "Missing required parameter: partition_size"
+        }
+        if (-not $size_is_maximum) {
+            try {
+                $max_supported_size = (Get-Disk -Number $disk_number).LargestFreeExtent
+            } catch {
+                Fail-Json -obj $result -message "Unable to get maximum size supported by disk: $($_.Exception.Message)"
+            }
+
+            if ($ansible_partition_size -gt $max_supported_size) {
+                Fail-Json -obj $result -message "Partition size is not supported by disk. Use partition_size: -1 to get maximum size"
+            }
+        }
+
+        $supp_part_type = (Get-Disk -Number $disk_number).PartitionStyle
+        if ($null -ne $mbr_type) {
+            if ($supp_part_type -eq "MBR" -and $mbr_styles.ContainsKey($mbr_type)) {
+                $partition_style = $mbr_styles.$mbr_type
+            } else {
+                Fail-Json -obj $result -message "Incorrect partition style specified"
+            }
+        }
+        if ($null -ne $gpt_type) {
+            if ($supp_part_type -eq "GPT" -and $gpt_styles.ContainsKey($gpt_type)) {
+                $partition_style = $gpt_styles.$gpt_type
+            } else {
+                Fail-Json -obj $result -message "Incorrect partition style specified"
+            }
+        }
+
+        if (-not $check_mode) {
+            $ansible_partition = New-AnsiblePartition -DiskNumber $disk_number -Letter $drive_letter -SizeMax $size_is_maximum -Size $ansible_partition_size -MbrType $mbr_type -GptType $gpt_type -Style $partition_style
+        }
+        $result.changed = $true
+    }
+}
+
+if ($state -eq "present" -and $ansible_partition) {
+    if ($offline -NotIn ($null, $ansible_partition.IsOffline)) {
```

Comment:
Same thing here if `$offline` is `$null` should we be touching this?

Reply:
I'm not sure what you  mean.

Unless specifically declared, it doesn't do anything and the partition is created as per the Windows defaults.

Reply:
Yep sorry at the time I was reading it as do this when it was `$null` but that isn't the case.